indexer/src/main/java/org/pdxfinder/dataexport/UniversalDataExporter.java

```diff
@@ -113,74 +114,60 @@ public void export(String exportDir) throws IOException {
         XSSFWorkbook mutationWorkbook = getWorkbook(templateDir+"/mutation_template.xlsx");
         XSSFWorkbook cnaWorkbook = getWorkbook(templateDir+"/cna_template.xlsx");
 
-
-        if(metadataWorkbook != null){
+        if (metadataWorkbook != null && noMetaDataSheetsAreNull()) {
             updateSheetWithData(metadataWorkbook.getSheetAt(1), patientSheetDataExport, 6, 2);
-            updateSheetWithData(metadataWorkbook.getSheetAt(2), patientTumorSheetDataExport,6, 2);
-            updateSheetWithData(metadataWorkbook.getSheetAt(3), pdxModelSheetDataExport,6, 2);
+            updateSheetWithData(metadataWorkbook.getSheetAt(2), patientTumorSheetDataExport, 6, 2);
+            updateSheetWithData(metadataWorkbook.getSheetAt(3), pdxModelSheetDataExport, 6, 2);
             updateSheetWithData(metadataWorkbook.getSheetAt(4), pdxModelValidationSheetDataExport, 6, 2);
-            updateSheetWithData(metadataWorkbook.getSheetAt(5), sharingAndContactSheetDataExport, 6, 2 );
+            updateSheetWithData(metadataWorkbook.getSheetAt(5), sharingAndContactSheetDataExport, 6, 2);
             updateSheetWithData(metadataWorkbook.getSheetAt(6), loaderRelatedDataSheetDataExport, 6, 2);
         }
 
-        if(samplePlatformWorkbook != null){
+        if (samplePlatformWorkbook != null && samplePlatformDescriptionSheetDataExport != null) {
             updateSheetWithData(samplePlatformWorkbook.getSheetAt(0), samplePlatformDescriptionSheetDataExport, 6, 1);
         }
 
-        if(mutationWorkbook != null){
-            updateSheetWithData(mutationWorkbook.getSheetAt(0), mutationSheetDataExport, 2, 1);
-        }
-
-        if(cnaWorkbook != null){
-            updateSheetWithData(cnaWorkbook.getSheetAt(0), cnaSheetDataExport, 2, 1);
-        }
-
-
-        // Write the output to a new file
-        FileOutputStream fileOut = null;
-
-        Path exportProviderDir = Paths.get(exportDir +"/"+ ds.getAbbreviation());
-        if (!exportProviderDir.toFile().exists()){
+        Path exportProviderDir = Paths.get(exportDir + "/" + ds.getAbbreviation());
+        if (!exportProviderDir.toFile().exists()) {
             Files.createDirectory(exportProviderDir);
         }
```

Comment:
No need to check the directory exists before creating if you use `createDirectories()` (see [https://stackoverflow.com/a/32454807](https://stackoverflow.com/a/3634879))
```suggestion
        Files.createDirectories(exportProviderDir);
```